### PR TITLE
docs: explain why ci-docs.yaml is needed

### PR DESCRIPTION
**Pull request by davidhsingyuchen**
## License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-# When a third-party action is added (i.e., `uses`), please also add it to `download-licenses` in Makefile.
+# - When a third-party action is added (i.e., `uses`), please also add it to `download-licenses` in Makefile.
+# - The set of jobs in this file should be equal to that in ci-docs.yaml.
```
Comment (ningziwen) [inline .github/workflows/ci.yaml:2]: Do we need to also doc in ci-docs.yaml that "The set of jobs in this file should be equal to that in ci.yaml."?

**Review by ningziwen — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (davidhsingyuchen) [reply .github/workflows/ci.yaml]: Good point. Fixed in c379793. Also updated the comments in `ci.yaml` to be clearer when one needs to update `ci-docs.yaml`.

File: .github/workflows/ci-docs.yaml [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# This file is created to address the issue described in
```
Comment (ahsan-z-khan) [inline .github/workflows/ci-docs.yaml:1]: Why do we need to link a stack overflow issue here? It's coming from GitHub's own documents. I think that would suffice.

**Review by ahsan-z-khan — CHANGES_REQUESTED** [review]

**Review by weikequ — DISMISSED** [review]
Great idea! Lgtm.

File: .github/workflows/ci-docs.yaml (hunk 2 above)
Comment (weikequ) [reply .github/workflows/ci-docs.yaml:1]: @davidhsingyuchen for context: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

Comment (davidhsingyuchen) [reply .github/workflows/ci-docs.yaml]: @weikequ Thanks a lot for the context! @ahsan-z-khan Fixed in [48e0498](https://github.com/runfinch/finch/pull/28/commits/48e0498635d9c3fb5c28de573c8ec82733987a77).

Comment (davidhsingyuchen) [reply .github/workflows/ci-docs.yaml]: Resolving this thread due to https://github.com/runfinch/finch/pull/28#pullrequestreview-1190879386.

**Pull request merged**